src/main/java/com/palantir/docker/compose/configuration/DockerType.java

```diff
@@ -15,20 +15,41 @@
  */
 package com.palantir.docker.compose.configuration;
 
-import com.google.common.base.StandardSystemProperty;
+import com.google.common.base.Optional;
+import java.util.Map;
 
-public enum DockerType {
+public enum DockerType implements HostIpResolver, EnvironmentValidator {
```

Comment:
Open to different approach for this part. Basic thing was that each DockerType had resolver and validator behavior associated with it, so wanted to encapsulate in code rather than having the current looser packaging.

Alternatives here are to use a class hierarchy instead (could just have a `DockerType` abstract class that defines `validate` and `resolveIp` and have subclasses provide implementation instead of breaking out into pieces like this) or to not have `DockerType` implement the interfaces but just provide getters that return the interfaces (although I think that just adds a layer of indirection without adding much benefit).